utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,254 @@
+          attachmentImageView.setImage(attachmentImage.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided attachmentSymbol").show();
+        }
+      });
+  
+      // add the connectivity symbol to the image view to display it in the legend
+      ListenableFuture<Image> connectivityImage = connectivitySymbol.createSwatchAsync(0x00000000);
+      connectivityImage.addDoneListener(() -> {
+        try {
+          // display the image in the image view
+          connectivityImageView.setImage(connectivityImage.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided connectivitySymbol").show();
+        }
+      });
+  
+      // create a unique value for each association type with the created symbols
+      UniqueValueRenderer.UniqueValue attachmentUniqueValue = new UniqueValueRenderer.UniqueValue(
+        "Attachment", "", attachmentSymbol, Collections.singletonList(UtilityAssociationType.ATTACHMENT.toString()));
+      UniqueValueRenderer.UniqueValue connectivityUniqueValue = new UniqueValueRenderer.UniqueValue(
+        "Connectivity", "", connectivitySymbol, Collections.singletonList(UtilityAssociationType.CONNECTIVITY.toString()));
+  
+      // create a unique value renderer using the values
+      UniqueValueRenderer uniqueValueRenderer = new UniqueValueRenderer(
+        Collections.singletonList("AssociationType"), Arrays.asList(attachmentUniqueValue, connectivityUniqueValue), "", null);
+      
+      // create a graphics overlay for the associations and set the renderer
+      associationsOverlay = new GraphicsOverlay();
+      associationsOverlay.setRenderer(uniqueValueRenderer);
+  
+      // create a map with the topographic vector basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createTopographicVector());
+      
+      // create a viewpoint to focus on the utility networks' extent
+      Viewpoint viewPoint = new Viewpoint(41.8057655, -88.1489692, 23);
+  
+      // set the initial viewpoint
+      map.setInitialViewpoint(viewPoint);
+      
+      // create a map view, set the map, and add the graphics overlay
+      mapView = new MapView();
+      mapView.setMap(map);
+      mapView.getGraphicsOverlays().add(associationsOverlay);
+      
+      // create the utility network
+      utilityNetwork = new UtilityNetwork("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer");
+  
+      // load the utility network and get all of the edges and junctions in the network
+      utilityNetwork.loadAsync();
+      utilityNetwork.addDoneLoadingListener(() -> {
+        List<UtilityNetworkSource> networkSources = utilityNetwork.getDefinition().getNetworkSources();
+  
+        for (UtilityNetworkSource nSource : networkSources)
```

Comment:
this would be better named as `networkSource` for purer clarity and readability
```suggestion
        for (UtilityNetworkSource networkSource : networkSources)
```